content/en/blog/_posts/2020-03-30-topology-manager-beta.md

```diff
@@ -0,0 +1,517 @@
+
+
+### HintProviders
+
+A **<code>HintProvider</code>** is a component internal to the <strong><code>kubelet</code></strong> that coordinates aligned resource allocations with the <strong><code>TopologyManager</code></strong>. At present, the only <strong><code>HintProviders</code></strong> in Kubernetes are the <strong><code>CPUManager</code></strong> and the <strong><code>DeviceManager</code></strong>. We plan to add support for <strong><code>HugePages</code></strong> soon.
+
+As discussed previously, the **<code>TopologyManager</code>** both gathers <strong><code>TopologyHints</code></strong> from <strong><code>HintProviders</code></strong> as well as triggers aligned resource allocations on them using a merged “best” hint. As such, <strong><code>HintProviders</code></strong> implement the following interface:
+
+
+```
+type HintProvider interface {
+    GetTopologyHints(*v1.Pod, *v1.Container) map[string][]TopologyHint
+    Allocate(*v1.Pod, *v1.Container) error
+}
+```
+
+
+Notice that the call to **<code>GetTopologyHints()</code> **returns a <strong><code>map[string][]TopologyHint</code></strong>. This allows a single <strong><code>HintProvider</code></strong> to provide hints for multiple resource types instead of just one. For example, the <strong><code>DeviceManager</code></strong> requires this in order to pass hints back for every resource type registered by its plugins.
+
+As **<code>HintProviders</code>** generate their hints, they only consider how alignment could be satisfied for <em>currently</em> available resources on the system. Any resources already allocated to other containers are not considered.
+
+For example, consider the system in Figure 1, with the following two containers requesting resources from it:
+
+
+<table>
+  <tr>
+   <td><strong><code>Container0</code></strong>
+   </td>
+   <td><strong><code>Container1</code></strong>
+   </td>
+  </tr>
+  <tr>
+   <td>
+     
+  <code>
+  spec:
+    containers:
+     - name: numa-aligned-container0
+       image: alpine
+       resources:
+       limits:
+          cpu: 2
+          memory: 200Mi
+          gpu-vendor.com/gpu: 1
+          nic-vendor.com/nic: 1
+  </code>
+   </td>
+   <td>
+  
+  <code>
+  spec:
+    containers:
+     - name: numa-aligned-container1
+       image: alpine
+       resources:
+       limits:
+          cpu: 2
+          memory: 200Mi
+          gpu-vendor.com/gpu: 1
+          nic-vendor.com/nic: 1
+  </code>
+   </td>
+  </tr>
+</table>
+
+
+If **<code>Container0</code>** is the first container considered for allocation on the system, the following set of hints will be generated for the three topology-aware resource types in the spec.
+
+	               
+	cpu: {{01: True}, {10: True}, {11: False}}
```

Comment:
Can we indent the text `{{01: True}, {10: True}, {11: False}}` to line up with where the similar text starts in the lines below this.